File: lab8/Java/src/warehouse/equipments/Weapon.java
package warehouse.equipments;

import enums.Caliber;
import enums.MaterialType;
import enums.SupplyType;
import java.util.Random;
import request.SupplyRequestDetail;

public class Weapon extends Equipment implements Cloneable {
    Caliber caliber;
    public Weapon(int id, String name, Caliber caliber, int count)
    {
        super(id, name, MaterialType.WEAPON, count);
        this.caliber = caliber;
    }

    @Override
    public Boolean isMatches(SupplyRequestDetail detail) {
        return detail.getSupplyType() == SupplyType.WEAPON &&
               detail.getCaliber() == caliber;
    }

    public Caliber getCaliber() {
        return caliber;
    }

    @Override
    public Weapon clone() throws CloneNotSupportedException{
        var clone = (Weapon)super.clone();
        clone.id = new Random().nextInt(0,9999);
        return clone;
    }    
}

File: CourseProject/ILS_WPF/Models/Core/Equipments/Weapon.cs
using ILS_WPF.Models.Core.Enums;
using ILS_WPF.Models.Core.Requests;

namespace ILS_WPF.Models.Core.Equipments
{
    public class Weapon : Equipment
    {
        public Caliber Caliber { get; set; }

        public Weapon() { } // Конструктор для EF

        public Weapon(string name, Caliber caliber, int count)
        {
            Name = name;
            MaterialType = MaterialType.Weapon;
            Caliber = caliber;
            Count = count;
        }

        public override bool IsMatches(SupplyRequestDetail detail)
            => detail.MaterialType == MaterialType.Weapon &&
                   Caliber == detail.Caliber;
    }
}
 